torch/_dynamo/debug_utils.py

```diff
@@ -962,33 +958,43 @@ def dynamo_accuracy_minifier_backend(gm, example_inputs, compiler_name):
 
     from torch._dynamo.optimizations.backends import BACKENDS
 
+    # (1) Get the backend
     if compiler_name == "inductor":
         from torch._inductor.compile_fx import compile_fx
 
         compiler_fn = compile_fx
     else:
+        assert (
+            compiler_name in BACKENDS
+        ), f"Unknown compiler name {compiler_name} provided."
         compiler_fn = BACKENDS[compiler_name]
 
-    # Set the eval mode to remove randomness.
-    gm.eval()
+    # (2) Compile the graph
+    compiled_gm = compiler_fn(copy.deepcopy(gm), clone_inputs(example_inputs))
 
-    # Check Accuracy
-    if backend_accuracy_fails(gm, example_inputs, compiler_fn):
-        log.warning("Accuracy failed for the TorchDyanmo produced graph")
-        dump_state_fn = functools.partial(
-            dump_backend_state, compiler_name=compiler_name, check_accuracy=True
-        )
-        fails_fn = functools.partial(
-            backend_accuracy_fails,
-            compiler_fn=compiler_fn,
-        )
-        dump_state_fn(fx.GraphModule(gm, copy.deepcopy(gm.graph)), example_inputs)
-        minifier(
-            gm,
-            example_inputs,
-            module_fails=fails_fn,
-            dump_state=dump_state_fn,
-        )
-    else:
-        log.error("Input graph does not fail accuracy testing")
-    return gm
+    def backend_accuracy_eval_fwd(*args):
+        # Set the eval mode to remove randomness.
+        gm.eval()
```

Comment:
This looks like an unnecessary behavior change; previously gm was changed to eval() mode at compile time, now it is reset to eval time every runtime.

...arguably we shouldn't be changing the model to eval mode at all, doesn't this mean running with accuracy minifier changes the results of your model...

Reply:
Good point, let's go back to the prior way where we ensure we compile.